snow/consensus/snowball/nnary_snowflake.go

```diff
@@ -57,26 +66,34 @@ func (sf *nnarySnowflake) RecordPoll(count int, choice ids.ID) {
 		return
 	}
 
-	if count < sf.alphaConfidence {
-		sf.confidence = 0
+	// If I need to change my preference, record the new preference
+	// and reset all my confidence counters.
+	if choice != sf.Preference() {
+		clear(sf.confidence)
 		sf.nnarySlush.RecordSuccessfulPoll(choice)
```

Comment:
I know this doesn't matter (and tbh might be an efficiency improvement) but it feels weird not to call `nnarySlush. RecordSuccessfulPoll` in every case here... Might just be me though

Reply:
I don't feel strongly about this. I can't imagine it's a meaningful performance improvement, so will go ahead and update it in both places.

Reply:
Addressed and updated the corresponding comment in both places